### PR TITLE
Use unique filenames in Serialization tests

### PR DESCRIPTION
**Pull request by shikharbhardwaj**
Fix for #1052 

`FilterFileName()` does not handle the corner case of `inputString` being empty, but I think it should be fine here.

### CODE REVIEW

File: src/mlpack/tests/test_tools.hpp [hunk 1]
```diff
@@ -48,4 +48,18 @@ inline void CheckMatrices(const arma::Mat<size_t>& a,
     BOOST_REQUIRE_EQUAL(a[i], b[i]);
 }
 
+inline std::string FilterFileName(const std::string& inputString)
```
Comment (rcurtin) [inline src/mlpack/tests/test_tools.hpp]: Can you add a comment to mention what this function is for please? Even in the tests we should be sure we have enough documentation. :)

**Comment by mentekid** [comment]
I like it. Thanks for being so fast with this :)
